coderd/workspaceupdates.go

```diff
@@ -86,46 +80,48 @@ func (s *sub) handleEvent(ctx context.Context, event wspubsub.WorkspaceEvent, er
 	}
 
 	s.prev = latest
-	s.tx <- out
+	select {
+	case <-s.ctx.Done():
+		return
+	case s.ch <- out:
```

Comment:
If the context has been cancelled, either the subscription has been closed, or the DRPC stream context has been cancelled, which will then close the subscription.